src/main/java/uk/gov/companieshouse/api/accounts/transaction/TransactionManagerImpl.java

```diff
@@ -1,42 +1,176 @@
 package uk.gov.companieshouse.api.accounts.transaction;
 
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.boot.web.client.RestTemplateBuilder;
 import org.springframework.http.HttpEntity;
 import org.springframework.http.HttpHeaders;
 import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
 import org.springframework.http.ResponseEntity;
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
 import org.springframework.stereotype.Component;
 import org.springframework.web.client.RestTemplate;
+import uk.gov.companieshouse.api.accounts.Kind;
+import uk.gov.companieshouse.api.accounts.LinkType;
 
 @Component
 public class TransactionManagerImpl implements TransactionManager {
 
-    private RestTemplate transactionRestTemplate;
-
     @Autowired
     private TransactionServiceProperties configuration;
 
     private static final String ID_PARAMETER = "{id}";
+    private static final String X_REQUEST_ID = "X-Request-Id";
+
+    // represents the Authorization header name in the request
```

Comment:
Can you make this into a JavaDoc rather than a comment please?
```java
/* represents the Authorization header name in the request */
private static final String AUTHORIZATION_HEADER = "Authorization";
```

Reply:
done